pathology/hovernet/training_ignite.py

```diff
@@ -0,0 +1,404 @@
+import sys
+
+from functools import partial
+import logging
+import os
+import time
+from argparse import ArgumentParser
+import torch
+import numpy as np
+import pandas as pd
+import torch.distributed as dist
+from monai.data import DataLoader, partition_dataset, CacheDataset
+from monai.networks.nets import HoVerNet
+from monai.engines import SupervisedEvaluator, SupervisedTrainer, PrepareBatchExtraInput
+from monai.transforms import (
+    Activations,
+    AsDiscrete,
+    AsDiscreted,
+    Compose,
+    ScaleIntensityRanged,
+    CastToTyped,
+    Lambdad,
+    SplitDimd,
+    EnsureChannelFirstd,
+    ComputeHoVerMapsd,
+    CenterSpatialCropd,
+    FillHoles,
+    BoundingRect,
+    ThresholdIntensity,
+    GaussianSmooth,
+    RandFlipd,
+    RandRotate90d,
+    RandGaussianSmoothd,
+)
+from monai.handlers import (
+    MeanDice,
+    CheckpointSaver,
+    LrScheduleHandler,
+    StatsHandler,
+    TensorBoardStatsHandler,
+    TensorBoardImageHandler,
+    ValidationHandler,
+    from_engine,
+)
+from monai.utils import set_determinism, ensure_tuple, convert_to_tensor
+from monai.utils.enums import HoVerNetBranch
+
+from loss import HoVerNetLoss
+from net import HoVerNet
+from transforms import (
+    GenerateWatershedMaskd,
+    GenerateInstanceBorderd,
+    GenerateDistanceMapd,
+    GenerateWatershedMarkersd,
+    Watershedd,
+    GenerateInstanceContour,
+    GenerateInstanceCentroid,
+    GenerateInstanceType,
+    ComputeHoVerMapsd,
+)
+
+
+def create_log_dir(cfg, fold):
+    timestamp = time.strftime("%y%m%d-%H%M%S")
+    run_folder_name = (
+        f"{timestamp}_hovernet_bs{cfg['batch_size']}_ep{cfg['n_epochs']}_lr{cfg['lr']}_fold{fold}"
+    )
+    log_dir = os.path.join(cfg["logdir"], run_folder_name)
+    print(f"Logs and model are saved at '{log_dir}'.")
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+    return log_dir
+
+def split_dataset(cfg):
+    # using original split in the paper
+    info = pd.read_csv(os.path.join(cfg["root"], "patch_info.csv"))
+    file_names = np.squeeze(info.to_numpy()).tolist()
+    split_info = pd.read_csv(os.path.join(cfg["root"], "split_info.csv"))
+
+    indices, splits = [], []
+    for i in range(3):
+        fold_case = split_info.loc[split_info['Split'] == i+1, 'Filename'].tolist()
+        fold_patches = [
+            file_name for file_name in file_names for _name in fold_case if _name == file_name.split("-")[0]
+        ]
+        fold_patches = np.unique(fold_patches)
+        print(f"Fold: {i} - {len(fold_patches):04d}")
+
+        fold_indices = [file_names.index(v) for v in fold_patches]
+        indices.append(fold_indices)
+
+    for i in range(3):
+        _indices = indices.copy()
+        splits.append({
+            "valid": _indices.pop(i),
+            "train": sum(_indices, []),
+        })
+
+    return splits
```

Comment:
Is it something that we want to include in the bundle too?

Reply:
Hi @drbeh, since it is written for the Lizard dataset, maybe I will remove it in the latest commit. And for how to prepare the CoNSep dataset, maybe we can include it in the bundle like this.
https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation/scripts
Thanks!

Reply:
Hi @KumoLiu, sure, that would work in this case but in general it is better to avoid scripts where possible.